pyOCD/flash/flash_builder.py

```diff
@@ -252,6 +250,7 @@ def program(self, chip_erase=None, progress_cb=None, smart_flash=True, fast_veri
                 flash_operation = self._page_erase_program(progress_cb)
 
         self.flash.target.resetStopOnReset()
+        self.flash.init_called_num = 0
```

Comment:
why do we manage this counter from outside of the class?
what is the rule of thumb here? `self.flash.target.setTargetState("PROGRAM")` should be called once after `self.flash.target.resetStopOnReset()`? so why not calling it directly? It seems that `FlashBuilder` knows about `flash.target` already